app/services/ScanUploadedFilesFlow.scala

```diff
@@ -27,8 +27,10 @@ class ScanUploadedFilesFlow @Inject()(
   consumer: QueueConsumer,
   parser: MessageParser,
   scanningService: ScanningService,
-  scanningResultHandler: ScanningResultHandler)(implicit ec: ExecutionContext)
+  scanningResultHandler: ScanningResultHandler,
+  ec2InstanceTerminator: InstanceTerminator)(implicit ec: ExecutionContext)
     extends PollingJob {
+
   def run(): Future[Unit] = {
     val outcomes = for {
       messages        <- consumer.poll()
```

Comment:
How many massages can there be? In the next line you are instantiating a scan for all messages concurrently which may fail if the number of messages is sufficiently large

Reply:
It's 1 :) But I will explicitly add this to the code.